File: python/pali_test.py
#  run with below
#  python3 -m unittest pali_test.py

import pali
import unittest


class TestStringInput(unittest.TestCase):
    def test_str_isPalindrome(self):
        self.assertTrue(pali.isPalindrome("racecar"))
        self.assertTrue(pali.isPalindrome("abbaabba"))
        self.assertFalse(pali.isPalindrome("artbdtra"))


class TestArrayInput(unittest.TestCase):
    def test_list_isPalindrome(self):
        self.assertTrue(pali.isPalindrome([[1, 2, 3, 4, 3, 2, 1]]))
        self.assertTrue(pali.isPalindrome([5, 6, 7, 7, 6, 5]))
        self.assertFalse(pali.isPalindrome([8, 9, 3, 1, 2, 3, 9, 8]))


if __name__ == "__main__":
    unittest.main()

File: java/algods-java/src/test/java/com/example/langaaegeskov/PaliTest.java
// package com.example;
package com.example.langaaegeskov;

import static org.junit.Assert.assertTrue;
import static org.junit.Assert.assertFalse;

import org.junit.Test;

/**
 * Unit test for simple App.
 */
public class PaliTest {
    /**
     * test even odd # Palindrome 'racecar'
     */
    @Test
    public void test_01() {
        String possiblePali_01 = "racecar";
        assertTrue(Pali.isPalindrome(possiblePali_01));
    }

    @Test
    public void test_02() {
        String possiblePali_02 = "abbaabba";
        assertTrue(Pali.isPalindrome(possiblePali_02));
    }

    @Test
    public void test_03() {
        String possiblePali_03 = "artbdtra";
        assertFalse(Pali.isPalindrome(possiblePali_03));
    }
}
